### ISSUE TITLE
It is not a "docking" branch

### ISSUE THREAD

**Issue by triuk** [ZenSepiol/Dear-ImGui-App-Framework#4]
Hi I wrote it below your [Youtube video](https://www.youtube.com/watch?v=cpJZnDWgDrk&lc=UgxSEgrBXKMOdUt3Dnl4AaABAg.9mHElagj0AH9mHRlLu56Dp), but it got somehow erased. I have [git](https://gitlab.com/uklaad/imvtk) where I use the docking branch within your code you initially wrote for ImGui framework.
See lines [90,91](https://gitlab.com/uklaad/imvtk/-/blob/main/src/include/app_design.h#L90).
And former if statement (I had to remove it to make it work) at [174-177](https://gitlab.com/uklaad/imvtk/-/blob/main/src/include/app_design.h#L174).
This way I can use the docking features and have separate windows outside the main window like it should be.

**Comment by ZenSepiol** [comment]
TL;DR
**Will implement as soon as I find the time.**


Thanks for the issue! 

I think your comment is still in the video, at least I don't delete any comments there. 

If I understand you correctly: You want to use the docking branch of Dear ImGui. Currently, this framework is using the master branch https://github.com/ZenSepiol/Dockerized-Dear-ImGui/blob/5c72313da76fe96b9c520842b89405218a32e34a/Dockerfile#L13 

I can implement the changes to use the docking branch, shouldn't be to much work. Until I get the time to do so, you can always use 
```
ImGui::GetIO().ConfigFlags |= ImGuiConfigFlags_DockingEnable;
```  
inside the startup function of your App 
https://github.com/ZenSepiol/Dockerized-Dear-ImGui/blob/5c72313da76fe96b9c520842b89405218a32e34a/src/app.hpp#L10-L12

Remember to change the used branch of Dear ImGui in the Dockerfile though. 

As a side-node: I publish my code under MIT license (https://github.com/ZenSepiol/Dockerized-Dear-ImGui/blob/main/LICENSE.md). Please honor the license by at least including the copyright notice and the license if you copy large parts of the code.

**Comment by triuk** [comment]
Hi, I can not make your solution work with the
```
if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
        {
            GLFWwindow* backup_current_context = glfwGetCurrentContext();
            ImGui::UpdatePlatformWindows();
            ImGui::RenderPlatformWindowsDefault();
            glfwMakeContextCurrent(backup_current_context);
        }
```
in the main loop. I must erase the if condition, because there is link to `io.ConfigFlags ` which does not exist in this scope.

**And BTW why did you avoid the class function features like "final" in your current solution?**

**Comment by ZenSepiol** [comment]
Hi, 
you should be able to access the ConfigFlags inside the `Run` context by using:
```
auto io = ImGui::GetIO().ConfigFlags;
if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
...
```
Does this work?

The **final** and **override** keywords are not necessary, since the functions are not virtual anymore in the app_base class. e.g. https://github.com/ZenSepiol/Dockerized-Dear-ImGui/blob/main/src/lib/app_base/app_base.hpp#L183-L186 
There the `update` function uses CRTP to call the `update` function of the derived class (in this case the app itself). This avoids some runtime overhead since the compiler can already resolve the call during compile time.
Hope that helps.

**Comment by ZenSepiol** [comment]
Let me know if you find the time to test things out :-)
